server/src/main/java/org/candlepin/async/temp/AsyncJobResource.java

```diff
@@ -140,4 +145,20 @@ public AsyncJobStatusDTO getStatusAndDeleteIfFinished(
         return this.translator.translate(status, AsyncJobStatusDTO.class);
     }
 
+    @ApiOperation(notes = "Cancels a Job Status", value = "cancel")
+    @ApiResponses({ @ApiResponse(code = 400, message = ""), @ApiResponse(code = 404, message = "") })
+    @DELETE
+    @Path("/{job_id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public AsyncJobStatusDTO cancel(@PathParam("job_id") @Verify(AsyncJobStatus.class) String jobId) {
+        AsyncJobStatus status = this.jobCurator.get(jobId);
+        if (status.getState() == AsyncJobStatus.JobState.CANCELED) {
```

Comment:
There's an argument to be made that this check should be performed at the curator level, but disregarding that, we also don't want to clobber other terminal states. This can be paired down to something like:

```
if (status.getState().isTerminal()) {
  // throw exception
}
```

We also need to check that the status actually exists.

Reply:
Done.